arrows/core/mesh_intersect.cxx

```diff
@@ -0,0 +1,385 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief Implementation of mesh operations
+ */
+
+#include "mesh_intersect.h"
+#include <limits>
+#include <vital/logger/logger.h>
+
+namespace kwiver {
+
+namespace arrows {
+
+namespace core {
+
+using namespace kwiver::vital;
+
+/// Intersect the ray from point to a triangle
+bool
+mesh_intersect_triangle( const vital::point_3d& p,
+                         const vital::vector_3d& d,
+                         const vital::point_3d& a,
+                         const vital::point_3d& b,
+                         const vital::point_3d& c,
```

Comment:
I find it odd to be `using namespace kwiver::vital` but still have `vital::` here (and elsewhere).

Also, if each parameter is going to be on its own line, maybe consider breaking after the `(` so the parameters are only indented one level?